lib/stump.ex

```diff
@@ -47,6 +45,6 @@ defmodule Stump do
 
   @doc false
   def time() do
```

Comment:
```
def time(:test) do
 # your mock time
end

def time(_) do
 Stump.Time.DateTime
end
```

Reply:
I really don't like this approach, using interfaces with mock implementations that agree to a contract is a testable and structure approach used in many languages and is suggested here by the creator of elixir: http://blog.plataformatec.com.br/2015/10/mocks-and-explicit-contracts/

Reply:
We do not have access to the `:environment` variable when we import this into an application as it will be the `Application.get_env(:my_app, :environment)` and we have no way of knowing that before hand

Reply:
We could find a middle ground, where the interface approach is kept, but without the reliance on configs, by using a Mock library. 

https://github.com/bbc/ExMetrics/blob/master/test/test_helper.exs#L1